In [57]:
import pandas as pd
import numpy as np


In [49]:
df_books = pd.read_csv('/content/bestsellers-with-categories.csv', sep=',', header = 0)
df_books

Name  \
0                        10-Day Green Smoothie Cleanse   
1                                    11/22/63: A Novel   
2              12 Rules for Life: An Antidote to Chaos   
3                               1984 (Signet Classics)   
4    5,000 Awesome Facts (About Everything!) (Natio...   
..                                                 ...   
545       Wrecking Ball (Diary of a Wimpy Kid Book 14)   
546  You Are a Badass: How to Stop Doubting Your Gr...   
547  You Are a Badass: How to Stop Doubting Your Gr...   
548  You Are a Badass: How to Stop Doubting Your Gr...   
549  You Are a Badass: How to Stop Doubting Your Gr...   

                       Author  User Rating  Reviews  Price  Year        Genre  
0                    JJ Smith          4.7    17350      8  2016  Non Fiction  
1                Stephen King          4.6     2052     22  2011      Fiction  
2          Jordan B. Peterson          4.7    18979     15  2018  Non Fiction  
3               George Orwell          4.7    21424      6  2017      Fiction  
4    National Geographic Kids          4.8     7665     12  2019  Non Fiction  
..                        ...          ...      ...    ...   ...          ...  
545               Jeff Kinney          4.9     9413      8  2019      Fiction  
546               Jen Sincero          4.7    14331      8  2016  Non Fiction  
547               Jen Sincero          4.7    14331      8  2017  Non Fiction  
548               Jen Sincero          4.7    14331      8  2018  Non Fiction  
549               Jen Sincero          4.7    14331      8  2019  Non Fiction  

[550 rows x 7 columns]

In [36]:
df_books.head(2)

Name        Author  User Rating  Reviews  Price  \
0  10-Day Green Smoothie Cleanse      JJ Smith          4.7    17350      8   
1              11/22/63: A Novel  Stephen King          4.6     2052     22   

   Year        Genre  
0  2016  Non Fiction  
1  2011      Fiction

COMO ELIMINAR COLUMNAS (DROP COLUMNS)

In [38]:
df_books.drop('Genre',axis = 1).head(2)

Name        Author  User Rating  Reviews  Price  \
0  10-Day Green Smoothie Cleanse      JJ Smith          4.7    17350      8   
1              11/22/63: A Novel  Stephen King          4.6     2052     22   

   Year  
0  2016  
1  2011

Podemos observar a continuacion  no se borra del dataframe la categroia genre, solo lo borra en la salida.

In [39]:
df_books.head(2)

Name        Author  User Rating  Reviews  Price  \
0  10-Day Green Smoothie Cleanse      JJ Smith          4.7    17350      8   
1              11/22/63: A Novel  Stephen King          4.6     2052     22   

   Year        Genre  
0  2016  Non Fiction  
1  2011      Fiction

Para borrar del dataset completamente se debe utilizar la siguiente sintaxis

In [40]:
df_books.drop('Genre',axis = 1, inplace=True)

In [44]:
df_books.head(2)

Name        Author  User Rating  Reviews  Price  \
0  10-Day Green Smoothie Cleanse      JJ Smith          4.7    17350      8   
1              11/22/63: A Novel  Stephen King          4.6     2052     22   

   Year  
0  2016  
1  2011

otra manera de hacerlo es:

In [45]:
df_books = df_books.drop('Year',axis = 1 )

In [46]:
df_books.head(3)

Name              Author  User Rating  \
0            10-Day Green Smoothie Cleanse            JJ Smith          4.7   
1                        11/22/63: A Novel        Stephen King          4.6   
2  12 Rules for Life: An Antidote to Chaos  Jordan B. Peterson          4.7   

   Reviews  Price  
0    17350      8  
1     2052     22  
2    18979     15

esto tambien se puede con la funcion propia de python del

In [47]:
del df_books['Price']

In [48]:
df_books.head(2)

Name        Author  User Rating  Reviews
0  10-Day Green Smoothie Cleanse      JJ Smith          4.7    17350
1              11/22/63: A Novel  Stephen King          4.6     2052

COMO ELIMINAR FILAS [DROPE ROWS)

*reinicio la df_books
*Ahi que recordar que el uso de inplace =True es necesaria para el borrado definitivo al igual que con las columnas, a menos que por ejemplo la dataset df_books sera renombrado.

In [50]:
df_books.head(2)

Name        Author  User Rating  Reviews  Price  \
0  10-Day Green Smoothie Cleanse      JJ Smith          4.7    17350      8   
1              11/22/63: A Novel  Stephen King          4.6     2052     22   

   Year        Genre  
0  2016  Non Fiction  
1  2011      Fiction

In [51]:
df_books.drop(0,axis=0).head(2)

Name              Author  User Rating  \
1                        11/22/63: A Novel        Stephen King          4.6   
2  12 Rules for Life: An Antidote to Chaos  Jordan B. Peterson          4.7   

   Reviews  Price  Year        Genre  
1     2052     22  2011      Fiction  
2    18979     15  2018  Non Fiction

In [53]:
df_books.drop([0,1,2], axis=0).head(2)

Name  \
3                             1984 (Signet Classics)   
4  5,000 Awesome Facts (About Everything!) (Natio...   

                     Author  User Rating  Reviews  Price  Year        Genre  
3             George Orwell          4.7    21424      6  2017      Fiction  
4  National Geographic Kids          4.8     7665     12  2019  Non Fiction

In [54]:
df_books.drop(range(0,10), axis=0).head(2)

Name            Author  \
10                          A Man Called Ove: A Novel   Fredrik Backman   
11  A Patriot's History of the United States: From...  Larry Schweikart   

    User Rating  Reviews  Price  Year        Genre  
10          4.6    23848      8  2017      Fiction  
11          4.6      460      2  2010  Non Fiction

In [55]:
df_books.head(2)

Name        Author  User Rating  Reviews  Price  \
0  10-Day Green Smoothie Cleanse      JJ Smith          4.7    17350      8   
1              11/22/63: A Novel  Stephen King          4.6     2052     22   

   Year        Genre  
0  2016  Non Fiction  
1  2011      Fiction

ADD COLUMNS

In [56]:
df_books.head(2)

Name        Author  User Rating  Reviews  Price  \
0  10-Day Green Smoothie Cleanse      JJ Smith          4.7    17350      8   
1              11/22/63: A Novel  Stephen King          4.6     2052     22   

   Year        Genre  
0  2016  Non Fiction  
1  2011      Fiction

In [60]:
df_books['Nueva Columna'] = np.nan
df_books.head(2)

Name        Author  User Rating  Reviews  Price  \
0  10-Day Green Smoothie Cleanse      JJ Smith          4.7    17350      8   
1              11/22/63: A Novel  Stephen King          4.6     2052     22   

   Year        Genre  Nueva Columna  
0  2016  Non Fiction            NaN  
1  2011      Fiction            NaN

In [63]:
df_books.shape[0]

550

Ahora creo una lista con todos los numeros del 0 al 549 la asigno a una columna llamada 'Rango'. Python entiende esto siempre y cuando la lista que sera columan tenga las mismas dimenciones que el numero de elemnetos de cada columna.

In [68]:
data = np.arange(0, df_books.shape[0])


In [69]:
df_books['Rango'] = data

In [70]:
df_books

Name  \
0                        10-Day Green Smoothie Cleanse   
1                                    11/22/63: A Novel   
2              12 Rules for Life: An Antidote to Chaos   
3                               1984 (Signet Classics)   
4    5,000 Awesome Facts (About Everything!) (Natio...   
..                                                 ...   
545       Wrecking Ball (Diary of a Wimpy Kid Book 14)   
546  You Are a Badass: How to Stop Doubting Your Gr...   
547  You Are a Badass: How to Stop Doubting Your Gr...   
548  You Are a Badass: How to Stop Doubting Your Gr...   
549  You Are a Badass: How to Stop Doubting Your Gr...   

                       Author  User Rating  Reviews  Price  Year        Genre  \
0                    JJ Smith          4.7    17350      8  2016  Non Fiction   
1                Stephen King          4.6     2052     22  2011      Fiction   
2          Jordan B. Peterson          4.7    18979     15  2018  Non Fiction   
3               George Orwell          4.7    21424      6  2017      Fiction   
4    National Geographic Kids          4.8     7665     12  2019  Non Fiction   
..                        ...          ...      ...    ...   ...          ...   
545               Jeff Kinney          4.9     9413      8  2019      Fiction   
546               Jen Sincero          4.7    14331      8  2016  Non Fiction   
547               Jen Sincero          4.7    14331      8  2017  Non Fiction   
548               Jen Sincero          4.7    14331      8  2018  Non Fiction   
549               Jen Sincero          4.7    14331      8  2019  Non Fiction   

     Nueva Columna  Rango  
0              NaN      0  
1              NaN      1  
2              NaN      2  
3              NaN      3  
4              NaN      4  
..             ...    ...  
545            NaN    545  
546            NaN    546  
547            NaN    547  
548            NaN    548  
549            NaN    549  

[550 rows x 9 columns]

AGRAGANDO FILAS
(ADD ROWS)

In [71]:
df_books.append(df_books)

Name  \
0                        10-Day Green Smoothie Cleanse   
1                                    11/22/63: A Novel   
2              12 Rules for Life: An Antidote to Chaos   
3                               1984 (Signet Classics)   
4    5,000 Awesome Facts (About Everything!) (Natio...   
..                                                 ...   
545       Wrecking Ball (Diary of a Wimpy Kid Book 14)   
546  You Are a Badass: How to Stop Doubting Your Gr...   
547  You Are a Badass: How to Stop Doubting Your Gr...   
548  You Are a Badass: How to Stop Doubting Your Gr...   
549  You Are a Badass: How to Stop Doubting Your Gr...   

                       Author  User Rating  Reviews  Price  Year        Genre  \
0                    JJ Smith          4.7    17350      8  2016  Non Fiction   
1                Stephen King          4.6     2052     22  2011      Fiction   
2          Jordan B. Peterson          4.7    18979     15  2018  Non Fiction   
3               George Orwell          4.7    21424      6  2017      Fiction   
4    National Geographic Kids          4.8     7665     12  2019  Non Fiction   
..                        ...          ...      ...    ...   ...          ...   
545               Jeff Kinney          4.9     9413      8  2019      Fiction   
546               Jen Sincero          4.7    14331      8  2016  Non Fiction   
547               Jen Sincero          4.7    14331      8  2017  Non Fiction   
548               Jen Sincero          4.7    14331      8  2018  Non Fiction   
549               Jen Sincero          4.7    14331      8  2019  Non Fiction   

     Nueva Columna  Rango  
0              NaN      0  
1              NaN      1  
2              NaN      2  
3              NaN      3  
4              NaN      4  
..             ...    ...  
545            NaN    545  
546            NaN    546  
547            NaN    547  
548            NaN    548  
549            NaN    549  

[1100 rows x 9 columns]